src/widgets.js

```diff
@@ -292,7 +296,69 @@ define([
         return widget;
     };
 
-    
+    var dropdown = function (mug, options) {
+        var widget = normal(mug, options);
+        widget.input = $("<select />")
+            .attr("name", widget.id)
+            .addClass('input-block-level');
+
+        var input = widget.input;
+
+        widget.setValue = function (value, option_name) {
+            var val = equivalentOption(value);
+            if (val) {
+                input.val(val.value);
+            } else {
+                widget.addOption(value, option_name || "");
+                input.val(value);
+            }
+        };
+
+        widget.getValue = function () {
+            return input.val();
+        };
+
+        widget.updateValue = function () {
+            widget.save();
+        };
+
+        input.change(function () {
+            widget.handleChange();
+        });
+
+        widget.addOption = function (value, text) {
+            var option = $('<option />')
+                .attr('value', value)
+                .text(text);
+            this.input.append(option);
+        };
+
+        widget.addOptions = function (options) {
+            var _this = this;
+            _.forEach(options, function(option) {
+                _this.addOption(option.value, option.text);
+            });
+        };
+
+        widget.getOptions = function () {
+            return _.map(widget.input.find('option'), function(option) {
+                return {
+                    value: option.value,
+                    text: option.text
```

Comment:
Neither of those will work because I'm iterating through each option. `attr("value")` and `text()` work for the first option if  I used `widget.option.text()` but you can't iterate through them.